gradoop-temporal/src/test/java/org/gradoop/temporal/model/impl/operators/snapshot/SnapshotTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.model.impl.operators.snapshot;
+
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.gradoop.temporal.model.api.functions.TemporalPredicate;
+import org.gradoop.temporal.model.impl.TemporalGraph;
+import org.gradoop.temporal.model.impl.functions.predicates.All;
+import org.gradoop.temporal.model.impl.functions.predicates.AsOf;
+import org.gradoop.temporal.model.impl.functions.predicates.Between;
+import org.gradoop.temporal.model.impl.functions.predicates.ContainedIn;
+import org.gradoop.temporal.model.impl.functions.predicates.CreatedIn;
+import org.gradoop.temporal.model.impl.functions.predicates.DeletedIn;
+import org.gradoop.temporal.model.impl.functions.predicates.FromTo;
+import org.gradoop.temporal.model.impl.functions.predicates.ValidDuring;
+import org.gradoop.temporal.util.TemporalGradoopTestBase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+/**
+ * Test for the snapshot operator for temporal graphs.
+ */
+@RunWith(Parameterized.class)
+public class SnapshotTest extends TemporalGradoopTestBase {
+
+  /**
+   * The temporal predicate to test.
+   */
+  @Parameterized.Parameter
+  public TemporalPredicate predicate;
+
+  /**
+   * The variable name of the input graph.
+   */
+  @Parameterized.Parameter(1)
+  public String inputGraph;
+
+  /**
+   * The expected result graph for the ascii graph loader.
+   */
+  @Parameterized.Parameter(2)
+  public String resultGraph;
+
+  /**
+   * Run the test. Calls the snapshot operator using a predicate and compares results with the
+   * expected result graph.
```

Comment:
Move line up

Reply:
This is kind of unnecessary since this does not fit, but oh well.